src/main/java/com/feywild/feywild/entity/Shroomling.java

```diff
@@ -0,0 +1,279 @@
+package com.feywild.feywild.entity;
+
+import com.feywild.feywild.FeywildMod;
+import com.feywild.feywild.entity.base.FeyBase;
+import com.feywild.feywild.entity.base.GroundFeyBase;
+import com.feywild.feywild.entity.base.ITameable;
+import com.feywild.feywild.entity.goals.GoToTargetPositionGoal;
+import com.feywild.feywild.entity.goals.SneezeGoal;
+import com.feywild.feywild.entity.goals.WaveGoal;
+import com.feywild.feywild.network.ParticleSerializer;
+import com.feywild.feywild.quest.Alignment;
+import net.minecraft.ChatFormatting;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.particles.SimpleParticleType;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.network.syncher.EntityDataAccessor;
+import net.minecraft.network.syncher.EntityDataSerializers;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.sounds.SoundEvent;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.util.Mth;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.MobSpawnType;
+import net.minecraft.world.entity.ai.attributes.AttributeSupplier;
+import net.minecraft.world.entity.ai.attributes.Attributes;
+import net.minecraft.world.entity.ai.control.MoveControl;
+import net.minecraft.world.entity.ai.goal.*;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.ComposterBlock;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.Tags;
+import software.bernie.geckolib3.core.IAnimatable;
+import software.bernie.geckolib3.core.PlayState;
+import software.bernie.geckolib3.core.builder.AnimationBuilder;
+import software.bernie.geckolib3.core.controller.AnimationController;
+import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
+import software.bernie.geckolib3.core.manager.AnimationData;
+import software.bernie.geckolib3.core.manager.AnimationFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.Random;
+
+public class Shroomling extends GroundFeyBase implements IAnimatable, ITameable {
+
+    public static final EntityDataAccessor<Integer> STATE = SynchedEntityData.defineId(DwarfBlacksmith.class, EntityDataSerializers.INT);
+    private static final double MIN_MOVING_SPEED_SQR = 0.05 * 0.05;
+    private final AnimationFactory factory = new AnimationFactory(this);
+    private boolean isTamed;
```

Comment:
Set this to false explicitly here or in the constructor as a non set field looks like the value doesn't matter until it is first set.